composeApp/src/main/kotlin/io/github/couchtracker/utils/ApiLoadableItemViewModel.kt

```diff
@@ -0,0 +1,93 @@
+package io.github.couchtracker.utils
+
+import android.app.Application
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.flow.transformLatest
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * A [androidx.lifecycle.ViewModel] specialized for loading data about an item asynchronously.
+ */
+abstract class ApiLoadableItemViewModel<T>(
```

Comment:
It seems to be that this class is doing two things:
- Converting `Flow<ApiLoadable<T>>` to `State<ApiLoadable<T>>` via `loadable` and collecting all the states in `allLoadables`
- Calling the API, using the cache, returning a Flow

Couldn't this two behaviors be split in two different classes?

The first one could be generified so that it's not forcing `ApiLoadable` but a generic `T`.